crates/agglayer-aggregator-notifier/src/certifier/mod.rs

```diff
@@ -88,172 +94,174 @@ impl<PendingStore, L1Rpc> CertifierClient<PendingStore, L1Rpc> {
-
-            let signer = Address::new(*signer.as_fixed_bytes());
-            let multi_batch_header = state
-                .apply_certificate(&certificate, signer, l1_info_root)
-                .map_err(|source| CertificationError::Types { source })?;
-
-            // Perform the native PP execution
-            let _ = generate_pessimistic_proof(initial_state.clone(), &multi_batch_header)
-                .map_err(|source| CertificationError::NativeExecutionFailed { source })?;
-
-            info!(
-                "Successfully executed the native PP for the Certificate {}",
-                certificate_id
-            );
-
-            let request = ProofGenerationRequest {
-                initial_state: default_bincode_options()
-                    .serialize(&initial_state)
-                    .map_err(|source| CertificationError::Serialize { source })?,
-                batch_header: default_bincode_options()
-                    .serialize(&multi_batch_header)
-                    .map_err(|source| CertificationError::Serialize { source })?,
-            };
-
-            info!("Sending the Proof generation request to the agglayer-prover service...");
-            let prover_response: tonic::Response<ProofGenerationResponse> = prover_client
-                .generate_proof(request)
-                .await
-                .map_err(|error| {
-                    debug!("Failed to generate the p-proof: {:?}", error);
-                    if let Ok(error) = default_bincode_options()
-                        .deserialize::<agglayer_prover_types::Error>(error.details())
-                    {
-                        match error {
-                            agglayer_prover_types::Error::UnableToExecuteProver => {
-                                CertificationError::InternalError("Unable to execute prover".into())
-                            }
-                            agglayer_prover_types::Error::ProverFailed(error) => {
-                                CertificationError::InternalError(error)
-                            }
-                            agglayer_prover_types::Error::ProofVerificationFailed(error) => {
-                                CertificationError::ProofVerificationFailed { source: error }
-                            }
-                            agglayer_prover_types::Error::ExecutorFailed(source) => {
-                                CertificationError::NativeExecutionFailed { source }
-                            }
-                        }
-                    } else {
-                        warn!(
-                            "Failed to deserialize the error details coming from the prover: \
-                             {error:?}"
-                        );
+        let mut state = state.clone();
+        let (multi_batch_header, initial_state) =
+            self.witness_execution(&certificate, &mut state).await?;
+
+        info!(
```

Comment:
Out of context of this PR, but we could probably refactor a bit logging using canonical log lines - it would be easier for monitoring - https://stripe.com/blog/canonical-log-lines
In this case:
```
 info!(%certificate_id, "Successfully executed the native PP");
```

Reply:
I think there is already a ticket to improve the logging system in the backlog :)